packages/flutter/lib/src/material/dropdown_theme.dart

```diff
@@ -0,0 +1,238 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'input_decorator.dart';
+import 'material_state.dart';
+import 'theme.dart';
+
+/// Used with [DropdownButton] and [DropdownButtonFormField] to define default
+/// property values for descendant [DropdownButton] and [DropdownButtonFormField]
+/// widgets.
+///
+/// Descendant widgets obtain the current [DropdownThemeData] object
+/// using `DropdownTheme.of(context)`. Instances of
+/// [DropdownThemeData] can be customized with
+/// [DropdownThemeData.copyWith].
+///
+/// A [DropdownThemeData] is often specified as part of the
+/// overall [Theme] with [ThemeData.dropdownTheme].
+///
+/// All [DropdownThemeData] properties are `null` by default.
+/// When a theme property is null, the [DropdownButton] and [DropdownButtonFormField]
+/// will provide its own default based on the overall [Theme]'s textTheme and
+/// colorScheme. See the individual [DropdownButton] and [DropdownButtonFormField]
+/// properties for details.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+///  * [DropdownTheme] which overrides the default [DropdownTheme]
+///    of its [DropdownButton] and [DropdownButtonFormField] descendants.
+///  * [ThemeData.textTheme], text with a color that contrasts with the card
+///    and canvas colors.
+///  * [ThemeData.colorScheme], the thirteen colors that most Material widget
+///    default colors are based on.
+@immutable
+class DropdownThemeData with Diagnosticable {
+  /// Creates a [DropdownThemeData].
+  const DropdownThemeData ({
+    this.dropdownColor,
+    this.style,
+    this.iconColor,
+    this.focusColor,
+    this.menuMaxHeight,
+    this.borderRadius,
+    this.inputDecorationTheme,
+  });
+
+  /// Overrides the default value of [DropdownButton.dropdownColor] and
+  /// [DropdownButtonFormField<T>.dropdownColor].
+  final Color? dropdownColor;
+
+  /// Overrides the default value of [DropdownButton.style] and
+  /// [DropdownButtonFormField<T>.style].
+  final TextStyle? style;
+
+  /// Overrides the default value of [DropdownButton.iconEnabledColor],
+  /// [DropdownButton.iconDisabledColor], [DropdownButtonFormField<T>.iconEnabledColor]
+  /// and [DropdownButtonFormField<T>.iconDisabledColor].
```

Comment:
Oooo these looks like potential deprecations in the future to migrate to a material state property? @darrenaustin is that true?